cheroot/server.py

```diff
@@ -1866,8 +1868,27 @@ def prepare(self):  # noqa: C901  # FIXME
         self.ready = True
         self._start_time = time.time()
 
+    def _serve_unservicable(self):
+        """Serve connections we can't handle a 503."""
+        while self.ready:
+            conn = self._unservicable_conns.get()
+            if conn is None:
+                return
+            request = HTTPRequest(self, conn)
+            try:
+                request.simple_response('503 Service Unavailable')
+            except Exception as ex:
```

Comment:
Is there a way to make this only intercept narrow exceptions?

Reply:
Sounds good. I think it'd be a good idea to also add a code comment explaining the need for catch-all. Though, maybe it's something to implement via a `@contextlib.contextmanager` or `contextlib.suppress`, so that the thread protection is separate from expected exception handling.